cmd/juju/machine/upgradeseries_test.go

```diff
@@ -137,18 +149,26 @@ func (s *UpgradeSeriesSuite) TestPrepareCommandShouldAcceptAgree(c *gc.C) {
 func (s *UpgradeSeriesSuite) TestPrepareCommandShouldPromptUserForConfirmation(c *gc.C) {
 	ctx, err := s.runUpgradeSeriesCommandWithConfirmation(c, "y", machine.PrepareCommand, machineArg, seriesArg)
 	c.Assert(err, jc.ErrorIsNil)
-	confirmationMsg := fmt.Sprintf(machine.UpgradeSeriesConfirmationMsg, machineArg, seriesArg, machineArg, unitsString)
+	confirmationMsg := fmt.Sprintf(machine.UpgradeSeriesConfirmationMsg,
+		machineArg, seriesArg, machineArg, strings.Join(units, "\n"))
 	c.Assert(ctx.Stdout.(*bytes.Buffer).String(), gc.Equals, confirmationMsg)
 }
 
 func (s *UpgradeSeriesSuite) TestPrepareCommandShouldAcceptAgreeAndNotPrompt(c *gc.C) {
 	ctx, err := s.runUpgradeSeriesCommandWithConfirmation(c, "n", machine.PrepareCommand, machineArg, seriesArg, "--agree")
 	c.Assert(err, jc.ErrorIsNil)
-	confirmationMessage := "" //There is no confirmation message since the `--agree` flag is being used to avoid the prompt
-	finishedMessage := fmt.Sprintf(machine.UpgradeSeriesPrepareFinishedMessage, machineArg)
+
+	//There is no confirmation message since the `--agree` flag is being used to avoid the prompt.
+	confirmationMessage := ""
+
+	finishedMessage := ""
+	for _, unit := range units {
+		finishedMessage = finishedMessage + fmt.Sprintf(
+			"leadership frozen for application %q\n", strings.Split(unit, "/")[0])
+	}
+	finishedMessage = fmt.Sprintf(finishedMessage+machine.UpgradeSeriesPrepareFinishedMessage, machineArg)
 	displayedMessage := strings.Join([]string{confirmationMessage, finishedMessage}, "") + "\n"
-	c.Assert(ctx.Stderr.(*bytes.Buffer).String(), gc.Equals, displayedMessage)
-	c.Assert(ctx.Stderr.(*bytes.Buffer).String(), gc.Equals, displayedMessage)
+	c.Assert(ctx.Stderr.(*bytes.Buffer).String(), gc.Matches, displayedMessage)
```

Comment:
Why use `Matches`. Is displayed message a regex? I may be mistaken but I don't see `displayedMessage` having any expressions.

Reply:
I was part way through changing this, then went with another solution. I'll put it back to `equals`.